cachi2/core/models/sbom.py

```diff
@@ -93,6 +106,235 @@ def _unique_components(cls, components: list[Component]) -> list[Component]:
         return unique_sorted(components, by=lambda component: component.key())
 
 
+class SPDXPackageExternalRefReferenceLocatorURI(pydantic.BaseModel):
+    """SPDX Package External Reference with URI reference locator."""
+
+    referenceLocator: str
+
+    @pydantic.validator("referenceLocator")
+    @classmethod
+    def _validate_uri_reference_locator(cls, referenceLocator: str) -> str:
+        parsed = urlparse(referenceLocator)
+        if not (parsed.scheme and (parsed.path or parsed.netloc)):
+            raise ValueError("Invalid URI reference locator")
+        return referenceLocator
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
```

Comment:
Can't express enough how much I appreciate all the docs references in all model classes' doc strings, this will help a lot anyone who wishes to dig into our design in the future.